pkg/engine/context/context.go

```diff
@@ -97,3 +100,35 @@ func (ctx *Context) AddUserInfo(userRequestInfo kyverno.RequestInfo) error {
 	}
 	return ctx.AddJSON(objRaw)
 }
+
+// removes prefix 'system:serviceaccount:' and namespace, then loads only username
+func (ctx *Context) AddSA(userName string) error {
+	saPrefix := "system:serviceaccount:"
+	var sa string
+	saName := ""
+	if len(userName) <= len(saPrefix) {
+		sa = ""
+	} else {
+		sa = userName[len(saPrefix):]
+	}
+	// filter namespace
+	groups := strings.Split(sa, ":")
+	if len(groups) >= 2 {
+		glog.V(4).Infof("serviceAccount namespace: %s", groups[0])
+		glog.V(4).Infof("serviceAccount name: %s", groups[1])
+		saName = groups[1]
+	}
+
+	glog.Infof("Loading variable serviceAccount with value: %s", saName)
+	saObj := struct {
+		SA string `json:"serviceAccount"`
+	}{
+		SA: saName,
```

Comment:
Apparently that's the requirement, as when we refer to a Service Account we only need the name excluding the namespace.
example:
```yaml
        - kind: ServiceAccount
          name: "{{serviceAccount}}"
          namespace: nirmata

```

But checking if it will be helpful to have 2 variables for service account:
- serviceAccountName
- serviceAccountNamespace

Reply:
Updated

Reply:
Ok, if that's the requirement. 

I'm thinking about that the tenant admin can own multiple namespaces (i.e., B, C, D..), while the tenant admin's serviceaccount is created with the specific namespace (i.e., A), there would be some use cases, i.e., to create the rolebinding in namespace B, which binds to tenant admin's serviceaccount(namespace A). We could address this in the next release.

Reply:
we have 2 variables now : 
- serviceAccountName
- serviceAccountNamespace